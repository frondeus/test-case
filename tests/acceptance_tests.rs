#![cfg(test)]

mod acceptance {
    use itertools::Itertools;
    use std::env;
    use std::path::PathBuf;
    use std::process::Command;

    #[test]
    fn basic() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("basic"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn hamcrest_assertions() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("hamcrest_assertions"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn r#async() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("async"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }
}
