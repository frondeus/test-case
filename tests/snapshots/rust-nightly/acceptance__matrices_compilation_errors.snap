---
source: tests/acceptance_tests.rs
expression: output
---
   |
   |
   |
   |
   |
   |
   |
   |
   |                          ++++++++++++++++++++
   |                  ^^^
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^
   |                ^^
   |     ^^^^^^^^^^^^^^^^^^^
   |    ^^^^^^^^^^^^^^^^^^^ -----
   | --------------^^^^^^^^^^^--
   | arguments to this function are incorrect
   | |             expected `i8`, found `usize`
   | |             |
  --> src/lib.rs:13:18
  --> src/lib.rs:18:18
  --> src/lib.rs:23:16
  --> src/lib.rs:30:5
  --> src/lib.rs:38:15
  --> src/lib.rs:39:4
  |
  |             ^
 --> src/lib.rs:5:13
13 | #[test_matrix(1..END)]
18 | #[test_matrix(0..9_223_372_036_854_775_808)]
23 | #[test_matrix(1..)]
30 |     ; "Illegal comment"
38 | #[test_matrix(USIZE_CONST)]
38 | #[test_matrix(USIZE_CONST.try_into().unwrap())]
39 | fn wrong_argument_type(x: i8) {
5 |     ["one", 1, true,]
For more information about this error, try `rustc --explain E0308`.
error: All literal values must be of the same type
error: Comments are not allowed in #[test_matrix]
error: Range bounds can only be an integer literal
error: Unbounded ranges are not supported
error: could not compile `matrices_compilation_errors` (lib test) due to 6 previous errors
error: number too large to fit in target type
error[E0308]: mismatched types
help: you can convert a `usize` to an `i8` and panic if the converted value doesn't fit
